cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,164 @@
+    def test_authRequired(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieAoiTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@test.test", password="testpassword"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movies_list(self):
+        sample_movie()
+
+        res = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movies_list_filter_by_genres(self):
+        sample_movie()
+
+        genre_drama = sample_genre(name="Drama")
+        genre_horror = sample_genre(name="Horror")
+
+        movie_with_genres = sample_movie()
+        movie_with_genres.genres.add(genre_drama, genre_horror)
+
+        res = self.client.get(
+            MOVIE_URL, {"genres": f"{genre_drama.id},{genre_horror.id}"}
+        )
+        movies = Movie.objects.filter(
+            genres__id__in=[genre_horror.id, genre_drama.id]
+        ).distinct()
+        serializer = MovieListSerializer(movies, many=True)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movies_list_filter_by_actors(self):
+        sample_movie()
+
+        actor_joe = sample_actor(first_name="Joe", last_name="Biden")
+        actor_donald = sample_actor(first_name="Donald", last_name="Trump")
+
+        movie_with_actors = sample_movie()
+        movie_with_actors.actors.add(actor_joe, actor_donald)
+
+        res = self.client.get(
+            MOVIE_URL, {"actors": f"{actor_joe.id},{actor_donald.id}"}
+        )
+        movies = Movie.objects.filter(
+            actors__id__in=[actor_joe.id, actor_donald.id]
+        ).distinct()
+        serializer = MovieListSerializer(movies, many=True)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movies_list_filter_by_title(self):
+        sample_movie()
+        sample_movie(title="Alien")
+
+        res = self.client.get(MOVIE_URL, {"title": "Alien"})
+        movies = Movie.objects.filter(title__icontains="Alien").distinct()
+        serializer = MovieListSerializer(movies, many=True)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movies_retrive(self):
+        movie = sample_movie(title="Alien")
+        actor_1 = sample_actor(first_name="Sigurni", last_name="Viver")
+        actor_2 = sample_actor(first_name="Thomas", last_name="Skerritt")
+
+        genre_1 = sample_genre(name="Sci-Fi")
+        genre_2 = sample_genre(name="Horror")
+
+        movie.actors.add(actor_1, actor_2)
+        movie.genres.add(genre_1, genre_2)
+
+        res = self.client.get(reverse("cinema:movie-detail", args=(movie.id,)))
+        serializer = MovieDetailSerializer(movie)
+        self.assertEqual(res.data, serializer.data)
+
```

Comment:
The method name `test_create_movie_forbiden` has a typo. It should be `test_create_movie_forbidden`.